build_tools/benchmarks/export_benchmark_config.py

```diff
@@ -23,28 +23,53 @@
 # Add build_tools python dir to the search path.
 sys.path.insert(0, str(pathlib.Path(__file__).parent.with_name("python")))
 
+from typing import Callable, Dict, List
 import argparse
 import collections
 import dataclasses
 import json
 
 from benchmark_suites.iree import benchmark_collections
+from e2e_test_framework.definitions import common_definitions
 from e2e_test_framework import serialization
 from e2e_test_artifacts import iree_artifacts
 
+DeviceSpecMatcher = Callable[[common_definitions.DeviceSpec], bool]
+BENCHMARK_PRESET_DEVICE_SPEC_MATCHERS: Dict[str, DeviceSpecMatcher] = {
+    "x86_64":
+        lambda device_spec: device_spec.architecture.architecture == "x86_64",
+    "cuda":
+        lambda device_spec: device_spec.architecture.architecture == "cuda",
+    "android-cpu":
+        lambda device_spec:
+        (device_spec.architecture.type == common_definitions.ArchitectureType.
+         CPU and device_spec.host_environment.platform == "android"),
+    "android-gpu":
+        lambda device_spec:
+        (device_spec.architecture.type == common_definitions.ArchitectureType.
+         GPU and device_spec.host_environment.platform == "android"),
+}
+
 
 def parse_arguments():
   """Parses command-line options."""
 
+  def parse_and_strip_list_argument(arg) -> List[str]:
+    return [part.strip() for part in arg.split(",")]
+
   parser = argparse.ArgumentParser(
-      description="Exports JSON config for benchmarking.")
+      description="Exports JSON config for benchmarking. Filters can be "
+      "specified jointly to select a subset of benchmarks.")
   parser.add_argument(
-      "--target_device_name",
-      type=str,
-      action="append",
-      dest="target_device_names",
-      help=("Target device name, can be specified multiple times. "
-            "Not specified means including all devices."))
+      "--target_device_names",
+      type=parse_and_strip_list_argument,
```

Comment:
Not sure what's the best practice for this. Using multiple optional arguments or [nargs](https://docs.python.org/3/library/argparse.html#nargs) can avoid the parsing here. But also bash scripts have some troubles to pass arguments in that format.

For example, if we pass the benchmark presets in environment variables in the CI workflow, for bash the list of presets is just a string. In order to pass as multiple arguments to this export tool, bash needs to parse the string into a bash array, so results in something like
```sh
IFS=',' read -r -a PRESET_ARRAY <<< "${BENCHMARK_PRESETS}"
./export_benchmark_config.py \
  "${PRESET_ARRAY[@]/#/--benchmark_preset=}"
```

Reply:
Yeah this seems good to me. If you add `action="extend"` then multiple instances of the flag would get concatenated, which seems nice. Maybe that's too complicated though.

Reply:
Yeah, it seems a little overkill to me : )